### PR TITLE
Use Flatbuffers for Persistent State of Segment Store and Meta Index

### PR DESCRIPTION
**Pull request by lava**
We want to use a well-defined on-disk format for VAST, to enable stable upgrade paths for VAST users.

This PR adds the first step of this effort by changing the segment store and the meta index to use a flatbuffer layout for its persistent on-disk data.

In addition, some preparatory work for more complete flatbuffer integration in the future is included, such as the `Layout` and `Type` type definitions. 

Finally, a new `filesystem` actor is included, which is not strictly related to flatbufferization but will be used in follow-up work to centralize the file reads and writes when loading and storing flatbuffers.

NOTE: This PR adds a hard dependency on `flatbuffers` to VAST.

### CODE REVIEW

**Comment by tobim** [comment]
Can you squash my `fixup!` commit?

I'm going to check if the branch works with the pro version locally.

**Comment by dominiklohmann** [comment]
Should we rebase this onto current master?

**Comment by lava** [comment]
@dominiklohmann The last commit merges current master into the branch.

Rebasing is problematic with such a big branch: First of all it is pretty labor-intensive but more importantly it potentially adds new conflict resolutions to a ton of old commits, potentially introducing subtle bugs that cannot be reviewed and invalidating the existing reviews.

**Comment by dominiklohmann** [comment]
[ch17762]

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ Every entry has a category for which we use the following visual abbreviations:
 
 ## Unreleased
 
+- ⚠️  Flatbuffers are now a hard dependency for building and running VAST.
+  The segment store and archive now use a flatbuffers table for their
+  on-disk persistent state.
```
Comment (dominiklohmann) [inline CHANGELOG.md:16]:
```suggestion
- ⚠️ Flatbuffers are now a hard dependency for building and running VAST.
  The segment store and archive now use a flatbuffers table for their
  on-disk persistent state. [#972](https://github.com/tenzir/vast/pull/972)
```

Also, should you rebase the last commit: Please remove the trailing dot from the commit message ;)

Comment (mavam) [inline CHANGELOG.md:16]:
```suggestion
- ⚠️  [Flatbuffers](https://google.github.io/flatbuffers/) is now a required
  dependency for VAST. The archive and segment store use flatbuffers to
  describe and version their persistent state.
```

Comment (dominiklohmann) [reply CHANGELOG.md]: Here's yours and my change combined:

```suggestion
- ⚠️ [Flatbuffers](https://google.github.io/flatbuffers/) is now a required
  dependency for VAST. The archive and segment store use flatbuffers to
  describe and version their persistent state.
  [#972](https://github.com/tenzir/vast/pull/972)
```

**Pull request merged**